ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/messenger/messagelistpage/MessageListContainerFragmentTest.java

```diff
@@ -0,0 +1,403 @@
+    Mockito.verify(messageListViewMock).hasUserInteractedWithList();
+  }
+
+  @Test
+  public void testHideReplyBox() {
+    // Arrange
+    PowerMockito.when(messageListContainerFragment.hasPageLoaded()).thenReturn(true);
+    ReplyBoxContract.ConfigureView replyBoxMock = Mockito.mock(ReplyBoxContract.ConfigureView.class);
+    Whitebox.setInternalState(messageListContainerFragment, "mReplyBoxView", replyBoxMock);
+
+    // Act
+    messageListContainerFragment.hideReplyBox();
+
+    // Assert
+    Mockito.verify(messageListContainerFragment).hideReplyBox();
+    Mockito.verify(replyBoxMock).hideReplyBox();
+  }
+
+  @Test
+  public void testShowReplyBox() {
+    // Arrange
+    PowerMockito.when(messageListContainerFragment.hasPageLoaded()).thenReturn(true);
+    ReplyBoxContract.ConfigureView replyBoxMock = Mockito.mock(ReplyBoxContract.ConfigureView.class);
+    Whitebox.setInternalState(messageListContainerFragment, "mReplyBoxView", replyBoxMock);
+
+    // Act
+    messageListContainerFragment.showReplyBox();
+
+    // Assert
+    Mockito.verify(messageListContainerFragment).showReplyBox();
+    Mockito.verify(replyBoxMock).showReplyBox();
+  }
+
+  @Test
+  public void testFocusOnReplyBox() {
+    // Arrange
+    PowerMockito.when(messageListContainerFragment.hasPageLoaded()).thenReturn(true);
+    ReplyBoxContract.ConfigureView replyBoxMock = Mockito.mock(ReplyBoxContract.ConfigureView.class);
+    Whitebox.setInternalState(messageListContainerFragment, "mReplyBoxView", replyBoxMock);
+
+    // Act
+    messageListContainerFragment.focusOnReplyBox();
+
+    // Assert
+    Mockito.verify(messageListContainerFragment).focusOnReplyBox();
+    Mockito.verify(replyBoxMock).focusOnReplyBox();
+  }
+
+  @Test
+  public void testSetReplyBoxHintMessage() {
+    // Arrange
+    PowerMockito.when(messageListContainerFragment.hasPageLoaded()).thenReturn(true);
+    ReplyBoxContract.ConfigureView replyBoxMock = Mockito.mock(ReplyBoxContract.ConfigureView.class);
+    Whitebox.setInternalState(messageListContainerFragment, "mReplyBoxView", replyBoxMock);
+
+    // Act
+    messageListContainerFragment.setReplyBoxHintMessage(HINT_MESSAGE);
+
+    // Assert
+    Mockito.verify(messageListContainerFragment).setReplyBoxHintMessage(HINT_MESSAGE);
+    Mockito.verify(replyBoxMock).setReplyBoxHintText(HINT_MESSAGE);
+  }
+
+  @Test
+  public void testCollapseToolbar() {
+    // Arrange
+    PowerMockito.when(messageListContainerFragment.hasPageLoaded()).thenReturn(true);
+    MessengerToolbarContract.ConfigureView toolbarViewMock = Mockito.mock(MessengerToolbarContract.ConfigureView.class);
+    Whitebox.setInternalState(messageListContainerFragment, "mToolbarView", toolbarViewMock);
+
+    // Act
+    messageListContainerFragment.collapseToolbar();
+
+    // Assert
+    Mockito.verify(messageListContainerFragment).collapseToolbar();
+    Mockito.verify(toolbarViewMock).collapseToolbarView();
+  }
+
+}
```

Comment:
⚠️ This file does not end with a termination newline character. Consider adding a termination newline at the end of the file to avoid unexpected behavior of different applications that may consume this file.

The POSIX standards define a new line as follows
&gt; A sequence of zero or more non-newline characters plus a terminating <newline> character

**[More details](http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_206)**

## Possible solution for this case

Consider adding a termination `newline` character at the end of the file 
</newline> 

<a id="1370e8a0-4f1e-11e8-b712-a55fdd633c3d" href="none" codereview-uuid-reserved-tag />